pkg/cloudprovider/aws/controllers/infrastructure/provider.go

```diff
@@ -0,0 +1,119 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package infrastructure
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/multierr"
+	"k8s.io/client-go/util/workqueue"
+	"knative.dev/pkg/logging"
+
+	"github.com/aws/karpenter/pkg/cloudprovider/aws"
+)
+
+type Provider struct {
+	sqsProvider         *aws.SQSProvider
+	eventBridgeProvider *aws.EventBridgeProvider
+}
+
+func NewProvider(sqsProvider *aws.SQSProvider, eventBridgeProvider *aws.EventBridgeProvider) *Provider {
+	return &Provider{
+		sqsProvider:         sqsProvider,
+		eventBridgeProvider: eventBridgeProvider,
+	}
+}
+
+func (p *Provider) CreateInfrastructure(ctx context.Context) error {
```

Comment:
Optional simplification. 
```suggestion
func (p *Provider) Create(ctx context.Context) error {
```